pkg/sqlite/load.go

```diff
@@ -435,10 +435,26 @@ func (s *sqlLoader) AddPackageChannelsFromGraph(graph *registry.Package) error {
 		var previousNodeID int64
 
 		// first clear the current channel graph
-		err := truncChannelGraph(tx, channelName, graph.Name)
+		query := `SELECT channel_name from channel_entry where channel_name = ? and package_name = ? LIMIT 1`
+		getChannelEntry, err := tx.Prepare(query)
 		if err != nil {
-			errs = append(errs, err)
-			break
+			return err
+		}
+		defer getChannelEntry.Close()
+
+		rows, err := getChannelEntry.Query(channelName, graph.Name)
+		if err != nil {
+			return err
+		}
+		// just remove the channel_entry when it exists
+		defer rows.Close()
+		if rows.Next() {
+			err := truncChannelGraph(tx, channelName, graph.Name)
+			if err != nil {
+				errs = append(errs, err)
+				break
+			}
+			return err
```

Comment:
nit: err will be nil, but elsewhere we always explicitly return nil for the sunny-day path.  
Is it better to do 'return nil' here?